include/nudb/error_code.hpp

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright (c) 2015-2016 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef ERROR_CODE_HPP
+#define ERROR_CODE_HPP
+
+#include <boost/system/error_code.hpp>
```

Comment:
This creates an exception from an error code which one can throw.
You never do this in the library implementation.  So you don't need it.
Our examples don't use it.  So one doesn't need it in order to use the library.
Library users can included if they want, but this is really a question
of application design. If we were going to include it anywhere it would be
in the documentation.